subsys/mgmt/mcumgr/smp_shell.c

```diff
@@ -85,6 +85,13 @@ static int read_mcumgr_byte(struct smp_shell_data *data, uint8_t byte)
 		}
 	}
 
+	if (IS_ENABLED(CONFIG_SHELL_LOG_BACKEND)) {
+		const struct shell *const sh = shell_backend_uart_get_ptr();
+
+		/* Reactivating the shell_log_backend */
+		z_shell_log_backend_resume(sh->log_backend);
+	}
```

Comment:
What if the logging was already disabled? This code seems to enable it regardless if it was disabled by this code or not. The way we currently deal with this is we send "log halt" over the UART prior to using mcumgr. Also what sort of impact does this have performance wise vs without this code doing the log halt method, are you able to do a firmware update with both methods and measure the difference in time/speed?